### PR TITLE
[GPU] Init tensor.data when allocating inputs for string type

### PR DESCRIPTION
**Pull request by davidsnam-intel**
### Details:
 - In case the element type is string produces the segmentation fault when input data is an empty string, unless the each element of tensor.data is initialized.

### Tickets:
 - 148921

### CODE REVIEW

**Review by p-wysocki — COMMENTED** [review]
All plugins handle empty strings without throwing but GPU, and this PR adds an additional if check for every string on all plugins. It's a high level fix for a low level issue.

Still, it's a fix for an existing issue, so I approve under the condition that is an additional story or discussion on what the desired plugin behavior for empty string is and adjust GPU if necessary.

**Review by akuporos — CHANGES_REQUESTED** [review]
I'd also expect to see the fix in GPU plugin directly, not in Python API.
As it was described in the ticket by @p-wysocki, the problem is GPU-specific, CPU works well.

File: src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp [hunk 1]
```diff
@@ -592,6 +592,12 @@ void SyncInferRequest::allocate_input(const ov::Output<const ov::Node>& port, si
     auto element_type = port.get_element_type();
 
     m_user_inputs[input_idx] = { create_host_tensor(shape, element_type), TensorOwner::PLUGIN };
+    if (element_type == ov::element::string) {
```
Comment (wilson-seok) [inline src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp:595]: Please add comment for initialization

Comment (davidsnam-intel) [reply src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp:595]: added.

**Review by p-wysocki — APPROVED** [review]

**Pull request merged**